src/cdk-experimental/table-scroll-container/table-scroll-container.ts

```diff
@@ -0,0 +1,136 @@
+  private _footerSizes: (number|null|undefined)[] = [];
+
+  constructor(
+      @Optional() private readonly _directionality: Directionality,
+      elementRef: ElementRef<HTMLElement>,
+      @Inject(DOCUMENT) document: any) {
+    this._document = document;
+    this._element = elementRef.nativeElement;
+    this._uniqueClassName = `cdk-table-scroll-container-${++nextId}`;
+    this._element.classList.add(this.CLASS_NAME);
+    this._element.classList.add(this._uniqueClassName);
+  }
+
+  ngOnDestroy(): void {
+    // TODO: Use remove() once we're off IE11.
+    if (this._styleElement?.parentNode) {
+      this._styleElement.parentNode.removeChild(this._styleElement);
+      this._styleElement = undefined;
+    }
+  }
+
+  stickyColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._startSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyEndColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._endSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyHeaderRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._headerSizes = sizes;
+    this._updateVertical();
+  }
+
+  stickyFooterRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._footerSizes = sizes;
+    this._updateVertical();
+  }
+
+  private _updateHorizontal(): void {
+    const startMargin = computeMargin(this._startSizes);
+    const endMargin = computeMargin(this._endSizes);
+
+    if (startMargin === 0 && endMargin === 0) {
+      this._applyCss('horizontal', '');
+      return;
+    }
+
+    const direction = this._directionality ? this._directionality.value : 'ltr';
+    const leftMargin = direction === 'rtl' ? endMargin : startMargin;
+    const rightMargin = direction === 'rtl' ? startMargin : endMargin;
+
+    this._applyCss('horizontal', `0 ${rightMargin}px 0 ${leftMargin}px`);
+  }
+
+  private _updateVertical(): void {
+    const topMargin = computeMargin(this._headerSizes);
+    const bottomMargin = computeMargin(this._footerSizes);
+
+    if (topMargin === 0 && bottomMargin === 0) {
+      this._applyCss('vertical', '');
+      return;
+    }
+
+    this._applyCss('vertical', `${topMargin}px 0 ${bottomMargin}px`);
+  }
+
+  private _getStyleSheet(): CSSStyleSheet {
+    if (!this._styleElement) {
+      this._styleElement = this._document.createElement('style');
+      this._styleElement.appendChild(this._document.createTextNode(''));
+      this._document.head.appendChild(this._styleElement);
+    }
+
+    return this._styleElement.sheet as CSSStyleSheet;
+  }
+
+  private _applyCss(orientation: Orientation, value: string) {
+    let index = this._ruleIndexes.get(orientation);
+    if (index === undefined) {
+      if (!value) {
+        return;
+      }
+
+      index = this._indexSequence++;
+      this._ruleIndexes.set(orientation, index);
+    } else {
+      this._getStyleSheet().deleteRule(index);
+    }
+
+    const selector = `.${this._uniqueClassName}::-webkit-scrollbar-track:${orientation}`;
```

Comment:
It looks like all the functionality of this module hinges on the WebKit-specific scrollbar selectors which definitely doesn't cover all the browsers we have in our browser support list. I suspect that we'll start getting bug reports about it not working in all browsers, similar to the `AutofillMonitor`.

Reply:
This is true - it would gracefully degrade to the current behavior in other browsers. I would be careful to document this clearly. Unfortunately the scrollbar-related CSS features in Firefox are not adequate to reproduce this feature.

Reply:
WebKit and blink do make up _most_ of the audience, though, so I don't think that's a huge problem for this feature since the fallback is normal browser behavior. It would be good to add a comment here explicitly calling that out, though.

Reply:
Doesn't that defeat the purpose of us saying the following?

> The Angular Components team supports the most recent two versions of all major browsers: Chrome (including Android), Firefox, Safari (including iOS), and IE11 / Edge.

Reply:
My philosophy is that something additive like this is fine so long as the all of the browsers still function normally. In this case, it means the scrollbar behaves very slightly differently for IE11 and Firefox.